### DISCUSSION TITLE
PWA not working iPadOS 15.1

### DISCUSSION THREAD

**Discussion by tim-rohrer** [coder/code-server#4405] [Q&A]
I am finding if I try to use code-server as a PWA, the bottom status bar moves to the top of the screen and I don’t have any work window. I’ve deleted the PWA and added it back to the Home Screen multiple times.

Is anyone else experiencing this? Better, is anyone successfully using code-server as a PWA on their iPad. 

I do seem to be able to use it in Safari and Chrome without too many issues, so the problem seems to be limited to the PWA.

Code server is running in a `linuxserver/code-server` (V3.12.0) container behind `authelia`. 

Any suggestions for things I can try?

**Comment by lg-nineoneninethreenine** [comment]
I feel like another discussion was opened on this topic, does the same happen if you use Google Chrome? If it does(PWA bug exists), try Firefox, Opera, Vivaldi, get the PWA and try again. If it doesn't work, only @jsjoeio can help on this.

**Reply by tim-rohrer** [reply]
My search didn’t reveal another topic on this, but I may have had different key terms.

I don’t know how to generate a PWA using Chrome but I had tried that before making my post. I just tried Firefox and same finding. Do you know how to generate a PWA using Chrome or Firefox?

Thanks.

**Reply by lg-nineoneninethreenine** [reply]
Well, I guess making a shortcut will do on Chrome, for Firefox, idk man. Vivaldi has to be the same since it is a Chromium-based browser with some modifications, but idk again, try creating a shortcut.

**Reply by jsjoeio** [reply]
Argh! This is very strange. AFAIK, only installing the PWA via Safari works on iPad. Can you post a video? I wonder if I disabled something in Settings to prevent this 🤔

**Reply by lg-nineoneninethreenine** [reply]
the legend is back

I think making a shortcut on Chrome will work if not on Safari since I distinctly remember making an older code-server Railway shortcut for code-server via Vivaldi(chromium-based) and after a few weeks, I opened it and it opened like a native app. But since that deployment is deleted, it showed an error, but it did work. Maybe as you told above, it could be something in the iPad's settings. But, I don't know, I don't have an iPad.

**Reply by tim-rohrer** [reply]
I got lucky this morning. Sometimes when it goes blank, I’m not able to get the screen back. In this circumstance, when I tapped the screen again it returns, but as soon as I click a key or try to tap on the screen to activate the window, it goes white.


https://user-images.githubusercontent.com/7756865/139436003-28f4b857-8a9b-4f91-9d6d-77e4ec1b6dab.mov

**Reply by lg-nineoneninethreenine** [reply]
Hey @jsjoeio haven't we seen this somewhere?

**Reply by jsjoeio** [reply]
Thanks for the video @tim-rohrer! That is very odd. Okay, I'd like to see if we can isolate this. A couple things I'm hoping you're willing to try:
- can you see if this happens when using vscode.dev as a PWA? (If it does, it could be upstream or specific to your iPad)
- if you're not using Railway with code-server, can you try deploying code-server and using it that way? see [video](https://www.youtube.com/watch?v=IWfkkaY5fXs)

For the white screen, you can also try using cmd + r to refresh. That may fix it.

Those will help us troubleshoot further. 

@im-coder-lg I _think_ so. I wish I had a cheatsheet for iPad fixes lol that would help here. I can't recall though.

**Reply by lg-nineoneninethreenine** [reply]
Well, feel like I noticed this in my notifications feed. But I think @code-asher might have answered that.

**Comment by tim-rohrer** [comment]
Good idea to try vscode.dev. It does indeed do it with a PWA from there, and upon researching the vscode repo, this issue was reported a few days ago. It seems to have something to do with the floating keyboard, although I’m not entirely sure what. I played around with keyboard settings and it started working. A bit finicky perhaps, but working nonetheless. I’ll give it a try for a bit and see if it is usable.

Thanks for the tip @jsjoeio!

**Reply by jsjoeio** [reply]
No problem! Thanks for trying it out so quickly. Glad to hear you got it at least working. Let us know if you find out anything else!
